client/state/notices/middleware.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+import { truncate } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { successNotice, errorNotice } from 'state/notices/actions';
+import { getSitePost } from 'state/posts/selectors';
+import {
+	POST_DELETE_FAILURE,
+	POST_DELETE_SUCCESS,
+	POST_SAVE_SUCCESS
+} from 'state/action-types';
+
+export const OBSERVERS = {
+	[ POST_DELETE_FAILURE ]: ( action, dispatch, getState ) => {
+		const post = getSitePost( getState(), action.siteId, action.postId );
+
+		let message;
+		if ( post ) {
+			message = translate( 'An error occurred while deleting "%s"', {
+				args: [ truncate( post.title, { length: 24 } ) ]
+			} );
+		} else {
+			message = translate( 'An error occurred while deleting the post' );
+		}
+
+		dispatch( errorNotice( message ) );
+	},
+	[ POST_DELETE_SUCCESS ]: ( action, dispatch ) => {
+		dispatch( successNotice( translate( 'Post successfully deleted' ) ) );
+	},
+	[ POST_SAVE_SUCCESS ]: ( action, dispatch ) => {
+		let text;
+		switch ( action.post.status ) {
+			case 'trash':
+				text = translate( 'Post successfully moved to trash' );
+				break;
+
+			case 'publish':
+				text = translate( 'Post successfully published' );
+				break;
+		}
+
+		if ( text ) {
+			dispatch( successNotice( text ) );
+		}
+	}
```

Comment:
>but I find it easier to scan through and test

I'd agree it'd make the list more scannable, but when I see code like how you'd described, I conversely find that I have trouble following the logic because my eyes are forced to bounce between the function declaration and its usage in the object.

>Why would a unit test even need to know that the handling code is inside some bigger data structure?

This is a good point, but what would be the difference between what you suggest and testing on `OBSERVERS[ POST_SAVE_SUCCESS ]`?